### PR TITLE
[CSL-354] Server side tracking

### PR DESCRIPTION
**Pull request by sblaurock**
Adds tracking methods largely inspired by the tracking module from constructorio-client-javascript. The differences are that `userParameters` are passed at method invocation time rather than client invocation time.

As requests originate server side, usual browser request header values need to be supplied within `userParameters`. They are as follows:

`x-cnstrc-token` (supplied at client invocation time)
`X-Forwarded-For`
`Accept-Language`
`Referer`
`User-Agent`

Personalization parameters are also supplied as with other modules.

Tests are comprehensive and ensure all possible request headers are sent correctly for every tracking method.

457/457 tests passing.
98.78% coverage on new tracker module.
93.79% coverage overall.

Screen Shot 2021-06-30 at 9 24 20 PM

Screen Shot 2021-06-30 at 9 30 26 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,7 +26,7 @@
   },
   "homepage": "https://github.com/Constructor-io/constructorio-node#readme",
   "engines": {
-    "node": ">=8.3.0"
+    "node": ">=14.5.0"
```
Comment (sblaurock) [inline package.json:29]: We should chat about this - eventemitter wasn't added until a decently late version of Node.

File: spec/src/modules/tracker.js [hunk 2]
```diff
@@ -0,0 +1,4836 @@
...
+const helpers = require('../../mocha.helpers');
+
+chai.use(chaiAsPromised);
+chai.use(sinonChai);
+dotenv.config();
+
+const delayBetweenTests = 25;
+const testApiKey = process.env.TEST_API_KEY;
+
+describe.only('ConstructorIO - Tracker', () => {
```
Comment (sblaurock) [inline spec/src/modules/tracker.js:20]: Remove `only`.

File: src/modules/tracker.js [hunk 3]
```diff
@@ -0,0 +1,1174 @@
...
+   * @param {string} [userParameters.acceptLanguage] - Client accept language
+   * @returns {(true|Error)}
+   */
+  trackSessionStart(userParameters) {
+    const url = `${this.options.serviceUrl}/behavior?`;
+    const queryParams = { action: 'session_start' };
+
+    send.call(
+      this,
+      `${url}${applyParamsAsString(queryParams, userParameters, this.options)}`,
```
Comment (jjl014) [inline src/modules/tracker.js:190]: Can we pull this out into a `requestURL` variable like we do for the v2 endpoints and similarly do this for the other v1 endpoint calls?

**Review by crgee1 — APPROVED** [review]
Looks good as far as I can tell. Ran the test, coverage, and docs. All looks solid. Nicely done

**Review by jjl014 — COMMENTED** [review]
Lgtm, just left one comment. 👍

File: src/modules/tracker.js (hunk 3 above)
Comment (sblaurock) [reply src/modules/tracker.js:190]: Sure can!

**Pull request merged**